curve25519-dalek/src/lib.rs

```diff
@@ -10,7 +10,10 @@
 // - Henry de Valence <hdevalence@hdevalence.ca>
 
 #![no_std]
-#![cfg_attr(all(curve25519_dalek_backend = "simd", nightly), feature(stdsimd))]
+#![cfg_attr(
+    all(curve25519_dalek_backend = "simd", nightly),
+    feature(stdarch_x86_avx512)
+)]
```

Comment:
Perhaps this should consider `target_arch` when setting `stdarch_x86_avx512` in order to be a little more futureproof? Something like:

```suggestion
#![cfg_attr(
    all(curve25519_dalek_backend = "simd", nightly, any(target_arch = "x86", target_arch = "x86_64")),
    feature(stdarch_x86_avx512)
)]
```

Reply:
I believe this is covered in the build file. Lmk if I'm wrong

https://github.com/dalek-cryptography/curve25519-dalek/blob/ba737a379071191158bacfa6d138f6249b12fc09/curve25519-dalek/build.rs#L50-L52

Reply:
The issue would occur if we added other non-x86 SIMD backends, e.g. #457

Reply:
I see, bc in that case `is_capable_simd` ought to be updated to be `true` when on aarch64. Ok yeah let's add that constraint.